docs/17-porting-tips.md

```diff
@@ -0,0 +1,333 @@
+This avoids most of the possible errors due to missing important changes.
+
+If you still wish to ignore this advice, first read the
+<a href="./89-NEWS.md.html">NEWS</a> section about the breaking changes. This
+document assumes you did.
+
+Be warned, even if it looks like it's working after changing some lines, **it wont
+supports dynamic screen changes and will have many subtle bugs**. The changes
+mentioned below **are important for the stability of your window manager**.
+
+Here is a diff of the 3.5.9 `rc.lua` with the 4.0 one. All changes due to
+new features and new syntaxes have been removed. A `-` in front of the line
+correspond to content of the 3.5 `rc.lua` and `+` its replacement in 4.0.
+
+To test during the port, we recommend the `Xephyr` X11 server:
+
+    Xephyr :1 -screen 1280x800 &
+    DISPLAY=:1 path/to/awesome -c /path/to/rc.lua --search /path/to/awesome/lib
+
+Screens are now added and removed without reloading `rc.lua`. The wallpaper are
+now set in a signal callback.
+
+    --- {{{ Wallpaper
+    -if `beautiful.wallpaper` then
+    -    for s = 1, screen.count() do
+    -        `gears.wallpaper.maximized`(beautiful.wallpaper, s, true)
+    -    end
+    -end
+    --- }}}
+    -for s = 1, `screen.count`() do
+    +local function set_wallpaper(s)
+    +    -- Wallpaper
+    +    if `beautiful.wallpaper` then
+    +        local wallpaper = `beautiful.wallpaper`
+    +        -- If wallpaper is a function, call it with the screen
+    +        if type(wallpaper) == "function" then
+    +            wallpaper = wallpaper(s)
+    +        end
+    +        `gears.wallpaper.maximized`(wallpaper, s, true)
+    +    end
+    +end
+    +
+    +-- Re-set wallpaper when a screen's geometry changes (e.g. different resolution)
+    +`screen.connect_signal`("property::geometry", set_wallpaper)
+
+Tags need to be created for each screens, the old static initialization cannot
+work. Remove this section.
+
+    --- {{{ Tags
+    --- Define a tag table which hold all screen tags.
+    -tags = {}
+    -for s = 1, `screen.count`() do
+    -    -- Each screen has its own tag table.
+    -    tags[s] = awful.tag({ 1, 2, 3, 4, 5, 6, 7, 8, 9 }, s, layouts[1])
+    - end
+    --- }}}
+
+The textclock is now part of the `wibox` library, rename it.
+
+    -- {{{ Wibar
+     -- Create a textclock widget
+    -mytextclock = `awful.widget.textclock`()
+    +mytextclock = `wibox.widget.textclock`()
+
+
+Widgets were previously added to static global tables. This isn't going to
+behave correctly when screen are added and removed. Remove this section.
+
+     -- Create a wibox for each screen and add it
+     -mywibox = {}
+     -mypromptbox = {}
+     -mylayoutbox = {}
+     -mytaglist = {}
+     -mytasklist = {}
+
+Many functions have been converted to methods. The old functions are deprecated,
+they are still supported, but will be removed in the next release. Also note
+that the name of the tables is now in a local variables instead of the table
```

Comment:
I'm lazy right now, so no fixup commit: `in a local variables` -> `in local variables`.

Also, I do not understand this sentence. The name of the table is in a local variable? So if the table is called `foo`. its name in a local variable would mean `local table_name = "foo"`. Then I would access the table as `_G[table_name]`. Obviously this is wrong, but this is how I understand the sentence.